saas/views/download.py

```diff
@@ -421,3 +424,273 @@ def queryrow_to_columns(self, record):
+
+        return list(consolidated_data.values())
+
+    def queryrow_to_columns(self, record):
+        return [
+            self.encode(record[heading]) for heading in self.headings
+        ]
+
+
+class PlanMetricsDownloadView(CSVDownloadView, PlanMetricsMixin):
+    basename = 'planmetrics'
+
+    def set_plans_data(self, request):
+        plans, _ = self.get_data(request=Request(request))
+        self.plans = plans
+        self.headings = ['Date'] + [plan['title'] for plan in plans]
+
+    def get_headings(self):
+        return self.headings
+
+    def get_queryset(self):
+        csv_data = [
+            ['location'] + [plan['location'] for plan in self.plans],
+            ['is_active'] + ['true' if plan['is_active'] else 'false' for plan in self.plans]
+        ]
+
+        consolidated_data = {}
+        for plan in self.plans:
+            for date_value in plan['values']:
+                date_str = date_value[0]
+                value = date_value[1]
+                if date_str not in consolidated_data:
+                    consolidated_data[date_str] = [0] * len(self.plans)
+                index = [p['title'] for p in self.plans].index(plan['title'])
+                consolidated_data[date_str][index] = value
+
+        for date, values in consolidated_data.items():
+            row = [date] + values
+            csv_data.append(row)
+
+        return csv_data
+
+    def queryrow_to_columns(self, record):
+        return self.encode(record)
+
+    def get(self, request, *args, **kwargs):
+        self.set_plans_data(request)
+        return super().get(request, *args, **kwargs)
+
+
+class BalancesDueDownloadView(BalancesDueMixin, CSVDownloadView):
+    basename = 'balances_due'
+
+    def get_headings(self):
+        basic_headers = [
+            'Slug',
+            'Printable Name',
+            'Picture',
+            'Type',
+            'Credentials',
+            'Created At'
+        ]
+
+        currency_set = self.get_currency_set()
+        currency_headers = [
+            f'{currency}_{balance_type}'
+            for currency in currency_set
+            for balance_type in ['contract_value', 'cash_payments', 'balance']
+        ]
+
+        return basic_headers + currency_headers
+
+    def get_currency_set(self):
+        queryset = self.get_queryset()
+        decorated_queryset = self.decorate_queryset(queryset)
+        currency_set = set()
+
+        for organization in decorated_queryset:
+
+            for currency in organization.get('balances').keys():
+                currency_set.add(currency)
+        return currency_set
+
+    def get_queryset(self):
+        queryset = super(BalancesDueDownloadView, self).get_queryset()
+        decorated_queryset = super(BalancesDueDownloadView, self).decorate_queryset(
+            queryset)
+        # Using a serializer to get type, credentials
+        serializer = BalancesDueSerializer(decorated_queryset, many=True)
```

Comment:
Why is the serialization happening in `get_queryset`? That doesn't seem the correct place to do it.